main.go

```diff
@@ -97,25 +72,8 @@ func main() {
 }
 
 func applyCacheConfig(cacheConfig string) error {
```

Comment:
Please remove this function, this single check can easily go to the upper scope.

Reply:
It is much easier if it stays :)

But the real reason I do not want to move it, is if I move to main it will have either a nested  or a complex if statement. Also, if there will be new cache types, this method will be needed. The method itself is next to it's usage, so it is not a big deal.

Reply:
@lpusok @godrei could you please share your thoughts with us?

Reply:
Having small method is actually a good pattern (https://martinfowler.com/bliki/FunctionLength.html). Nested ifs are considered code smell, however one level of nested if is not a big issue, but overall I think the current is a good solution.

Reply:
I agree with @trapacska and @lpusok this is an unnecessary function wrapping, makes harder to follow the data flow and add unnecessary extra lines to the code.

You can avoid having nested if statement (however the solution mentioned by @trapacska is the most readable)

```
	if configs.CacheLevel == "all" && err := applyAllCache(); err != nil {
		fail("Could not apply caching: %s", err)
	}
```

Pls update your code.